pkg/policy/resolve.go

```diff
@@ -18,17 +18,16 @@ import (
 	"github.com/cilium/cilium/pkg/identity"
 	"github.com/cilium/cilium/pkg/identity/cache"
 	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/option"
 	"github.com/cilium/cilium/pkg/policy/api"
 	"github.com/cilium/cilium/pkg/policy/trafficdirection"
+
 	"github.com/sirupsen/logrus"
 )
 
-// EndpointPolicy is a structure which contains the resolved policy across all layers
-// (L3, L4, and L7).
-type EndpointPolicy struct {
-	// ID is the node-local identifier of this EndpointPolicy.
-	ID uint16
-
+// IdentityPolicy is a structure which contains the resolved policy across all
+// layers (L3, L4, and L7), but still determined in terms of EndpointSelectors.
+type IdentityPolicy struct {
```

Comment:
The more I think about this, the more I think that `SelectorPolicy` is a more appropriate name for this level of abstraction. It *is* the policy for a particular Identity, and we will generate it when an identity is used by a local endpoint.... however, changes to remote identities do not result in changes to this structure. Rather, changes to policy rule endpoint selectors for allowing interaction with network peers change this structure. Furthermore, the contributing aspect of policy to this structure is whether the policy rules *select* endpoints with this identity.

Naming as `SelectorPolicy` would avoid confusion on this front by putting more distance between identity interactions and this structure.